### PR TITLE
KOGITO-538: Add Keycloak security integration to spring boot - kogito…

### PR DESCRIPTION
**Pull request by nmirasch**
… use case

### CODE REVIEW

File: kogito-usertasks-with-security-oidc-springboot/src/main/java/org/kie/kogito/tests/LogEventPublisher.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Collection;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.kie.kogito.event.DataEvent;
+import org.kie.kogito.event.EventPublisher;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LogEventPublisher implements EventPublisher {
```
Comment (mswiderski) [inline kogito-usertasks-with-security-oidc-springboot/src/main/java/org/kie/kogito/tests/LogEventPublisher.java]: not really needed ... sort of left over after my tests so can be removed

**Review by mswiderski — APPROVED** [review]
just a tiny comment other than that looks really nice. Good to see the feature parity for both runtimes. Excellent work @nmirasch

File: kogito-usertasks-with-security-oidc-springboot/src/main/java/org/kie/kogito/tests/LogEventPublisher.java (hunk 1 above)
Comment (nmirasch) [reply kogito-usertasks-with-security-oidc-springboot/src/main/java/org/kie/kogito/tests/LogEventPublisher.java]: Thanks @mswiderski, removed! .. but now you mention the test.. They are broken regarding the keycloak server needed.. I suppose we need something similar to the KeycloakServerTestResource what we had in quarkus..Do you think makes sense to have the complete testing stuf in this use case?

**Comment by mswiderski** [comment]
depending how time consuming it is ... if it's not too much I'd say it is worth to have it tested with every change/release

**Comment by nmirasch** [comment]
@mswiderski I have changed the test to retrieve the token with the keycloak running.(The same as is started for running the app, nothing specific for test) as a quick fix for them

**Comment by nmirasch** [comment]
@mswiderski not sure this solution makes sense, it will break the tests if the keycloak is not running :(
I'll try to folllow a better approach for this. Thanks man!

**Comment by nmirasch** [comment]
@mswiderski I added some test for both quarkus and spring boot use cases. Both using that Generic container used just for testing. I think now it looks more complete.

**Comment by mswiderski** [comment]
@nmirasch very nice work, looks good to go for me.

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**